dirsrvtests/tests/suites/plugins/dna_interval_test.py

```diff
@@ -82,3 +96,94 @@ def test_dna_interval(topology_st):
     log.info("Test DNA interval assignment is working...")
     user.replace('uidNumber', '-1')
     assert user.get_attr_val_utf8_l('uidNumber') == '20'
+
+
+def test_dna_max_value(topology_st, dna_plugin):
+    """Test the dna max value works with dna interval
+
+    :id: cc979ea8-3cd0-4d52-af35-9cea7cf8cb5f
+    :setup: Standalone Instance
+    :steps:
+        1. Set dnaMaxValue, dnaNextValue, dnaInterval values to 100, 90, 50 respectively
+        2. Create user that trigger DNA to assign a value
+        3. Try to make an update to entry that triggers DNA again
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Operation should fail with OPERATIONS_ERROR
+    """
+    log.info("Make the config changes needed to test dnaMaxValue")
+    dna_plugin.replace_many(('dnaMaxValue', '100'), ('dnaNextValue', '90'), ('dnaInterval', '50'))
+    # Create user2
+    users = UserAccounts(topology_st.standalone, DEFAULT_SUFFIX)
+    log.info('Adding user2')
+    user = users.create(properties={
+        'sn': 'interval2',
+        'cn': 'interval2',
+        'uid': 'interval2',
+        'uidNumber': '-1',  # Magic regen value
+        'gidNumber': '222',
+        'givenname': 'interval2',
+        'homePhone': '0861234567',
+        'carLicense': '131D16674',
+        'mail': 'interval2@example.com',
+        'homeDirectory': '/home/interval2'})
+
+    # Verify DNA works
+    assert user.get_attr_val_utf8_l('uidNumber') == '90'
+
+    log.info("Make an update and verify it raises error as the new interval value is more than dnaMaxValue")
+    with pytest.raises(ldap.OPERATIONS_ERROR):
+        user.replace('uidNumber', '-1')
+
+
+@pytest.mark.parametrize('attr_value', ('0', 'abc', '2000'))
+def test_dna_interval_with_different_values(topology_st, dna_plugin, attr_value):
+    """Test the dna interval with different values
+
+    :id: 1a3f69fd-1d8d-4046-ba68-b6aa7cafbd37
+    :parametrized: yes
+    :setup: Standalone Instance
+    :steps:
+        1. Set dnaInterval value to 0
+        2. Set dnaInterval value to 'abc'
+        3. Set dnaInterval value to 2000 and dnaMaxValue to 1000
+        4. Create user that trigger DNA to assign a value
+        5. Try to make an update to entry that triggers DNA again when dnaInerval is greater than dnaMaxValue
+    :expectedresults:
+        1. Operation should fail with SERVER_DOWN
+        2. Operation should fail with INVALID_SYNTAX
+        3. Success
+        4. Success
+        5. Operation should fail with OPERATIONS_ERROR
+    """
+    log.info("Make the config changes needed to test dnaInterval")
+    if attr_value == '0':
+        with pytest.raises(ldap.SERVER_DOWN):
```

Comment:
Why server down ! does the test crash the server ?

Reply:
Yes it does when the value is 0 if this is not expected then it's a bug.

Reply:
Congratulation you found a nice one